src/combinatorics/simple_graph/regularity/uniform.lean

```diff
@@ -80,11 +85,90 @@ begin
   exact zero_lt_one,
 end
 
+variables {G}
+
+lemma exists_nonuniform_witnesses (h : ¬G.is_uniform ε s t) :
+  ∃ s', s' ⊆ s ∧ ∃ t', t' ⊆ t ∧ ↑s.card * ε ≤ s'.card ∧ ↑t.card * ε ≤ t'.card ∧
+    ε ≤ |G.edge_density s' t' - G.edge_density s t| :=
+by { rw is_uniform at h, push_neg at h, exact h }
+
+open_locale classical
+variables (G)
+
+/-- A pair of subsets witnessing the non-uniformity of `(s, t)`. If `(s, t)` is uniform, returns
+`(s, t)`. Witnesses for `(s, t)` and `(t, s)` don't necessarily match, hence the motivation to
+define `nonuniform_witness`. -/
+noncomputable def nonuniform_witnesses (ε : 𝕜) (s t : finset α) : finset α × finset α :=
+if h : ¬G.is_uniform ε s t
+  then ((exists_nonuniform_witnesses h).some, (exists_nonuniform_witnesses h).some_spec.2.some)
+  else (s, t)
+
+lemma left_nonuniform_witnesses_subset (h : ¬G.is_uniform ε s t) :
+  (G.nonuniform_witnesses ε s t).1 ⊆ s :=
+by { rw [nonuniform_witnesses, dif_pos h], exact (exists_nonuniform_witnesses h).some_spec.1 }
+
+lemma left_nonuniform_witnesses_card (h : ¬ G.is_uniform ε s t) :
+  (s.card : 𝕜) * ε ≤ (G.nonuniform_witnesses ε s t).1.card :=
+by { rw [nonuniform_witnesses, dif_pos h],
+  exact (exists_nonuniform_witnesses h).some_spec.2.some_spec.2.1 }
```

Comment:
Indentation? I think this is supposed to be aligned with `rw` (for this one and each below).

Reply:
Hmm... I have never seen that rule.

Reply:
It looks like there's no agreement here (it's not perfect, but this turns up examples both ways: `git grep -P -n5 'by {.*(?!}).,$'`).  Aligning seems like it'd improve readability, but that's up to you.